### PR TITLE
Fixed tests

### PR DESCRIPTION
**Pull request by muhaale**

### CODE REVIEW

File: tests/anmykh/test_assertion_error.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import pytest
+
+
+def test_attribute_error():
+    with pytest.raises(AssertionError):
+        assert 0 <= (5 == True) == True
```
Comment (kudinov-fedor) [inline tests/anmykh/test_assertion_error.py:6]: minimum code required:

```
    with pytest.raises(AssertionError):
        assert False

    with pytest.raises(AssertionError):
        assert 0

    with pytest.raises(AssertionError):
        assert []
```

File: tests/anmykh/test_attribute_error.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import pytest
+
+
+def test_attribute_error():
+    sad = (2, 1, 43, 1)
+    with pytest.raises(AttributeError):
+        sad.pop(0)
```
Comment (kudinov-fedor) [inline tests/anmykh/test_attribute_error.py:7]:
```
sad.pop
```

will already cause an error

File: tests/anmykh/test_is_string.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+def test_is_string():
+    assert isinstance("sad", str)
```
Comment (kudinov-fedor) [inline tests/anmykh/test_is_string.py:2]: no need to make 1 file for 1 test,   

group tests together , if they relate to same idea e.g.  (errors, conversion etc.)

File: tests/anmykh/test_syntax_error.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import pytest
+
+def test_syntax_error():
+    sad = {"abc": 123}
+    with pytest.raises(SyntaxError):
+        sad{"abc"}
```
Comment (kudinov-fedor) [inline tests/anmykh/test_syntax_error.py:6]: test never runs, 
you need to rewrite it

you fail on module level

Comment (kudinov-fedor) [reply tests/anmykh/test_syntax_error.py:6]: you can write test like this
it fails on import of incorrect module


```
# my_module.py

print "hello world"


# -----------------------------------
# test_errors.py
import pytest

def test_syntax_error():
    
    with pytest.raises(SyntaxError):
        import my_module
```

File: tests/anmykh/test_syntax_error.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+def test_syntax_error2():
+   False = 0
+   with pytest.raises(SyntaxError):
+       False
```
Comment (kudinov-fedor) [inline tests/anmykh/test_syntax_error.py:11]: error happens in line 9, but not in line 11

File: tests/anmykh/test_assertion_error.py (hunk 1 above)
Comment (muhaale) [reply tests/anmykh/test_assertion_error.py]: fixed

File: tests/anmykh/test_attribute_error.py (hunk 2 above)
Comment (muhaale) [reply tests/anmykh/test_attribute_error.py]: fixed

File: tests/anmykh/test_is_string.py (hunk 3 above)
Comment (muhaale) [reply tests/anmykh/test_is_string.py]: fixed

File: tests/anmykh/test_syntax_error.py (hunk 4 above)
Comment (muhaale) [reply tests/anmykh/test_syntax_error.py]: fixed

File: tests/anmykh/test_syntax_error.py (hunk 5 above)
Comment (muhaale) [reply tests/anmykh/test_syntax_error.py]: fixed

File: tests/anmykh/test_homework_3.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def test_empty():
+    assert int() == 0
+    assert bool() == False
+    assert float() == 0.0
+    assert str() == ''
+    assert tuple() == ()
+    assert list() == []
+    assert range(10) == range(0, 10)
```
Comment (kudinov-fedor) [inline tests/anmykh/test_homework_3.py:12]: also 
assert range(10) == range(0, 10, 1).  # start, stop, step

File: tests/anmykh/test_homework_3.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    assert dict() == {}
+
+
+def test_is():
+    a = (1, 2, 3)
+    b = (1, 2, 3)
+    assert a == b
+    assert a is b
+    hash(a) == hash(b)
+    id(a) == id(b)
```
Comment (kudinov-fedor) [inline tests/anmykh/test_homework_3.py:23]: lines 21 and 23 should fail

File: tests/anmykh/test_errors.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+def test_type_error():
+    with pytest.raises(Exception):
+        assert 10 / "sad"
+
+
+def test_arithmetic_error():
+    with pytest.raises(Exception):
+        assert 10 / 0
+
```
Comment (kudinov-fedor) [inline tests/anmykh/test_errors.py:45]: better not to use too wide exceptions,
but for practice its ok

Comment (muhaale) [reply tests/anmykh/test_errors.py:45]: a good advice

File: tests/anmykh/test_first_homework.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+
+def test_is_set():
+    assert isinstance({3, 2, "sad"}, set)
+
+
+def test_is_tuple():
+    assert isinstance((12,5), tuple)
+
+    assert (12) is not tuple
```
Comment (kudinov-fedor) [inline tests/anmykh/test_first_homework.py:12]: incorrect check


a = (12)
assert type(a) is not tuple

File: tests/anmykh/test_homework_3.py [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+def test_empty():
+    assert int() == 0
+    assert bool() == False
+    assert float() == 0.0
+    assert str() == ''
+    assert tuple() == ()
+    assert list() == []
+    assert range(10) == range(0, 10, 1)
+    assert set() is not ()
```
Comment (kudinov-fedor) [inline tests/anmykh/test_homework_3.py:13]: incorrect check 
is - identity check
you need to compare values or types

set() != ()


a = ()
type(a) is not set

File: anmykh/checkio/checkio_2.py [hunk 11]
```diff
@@ -0,0 +1,11 @@
+#  introduce two new variables named add, multi.
+#  Assign them values equal to the sum and product, respectively, of the previously declared variables a, b.
+import operator
+
+
+def add_and_multi():
+    a = 2
+    b = 5
+    add = a + b
+    multi = operator.mul(a, b)
```
Comment (kudinov-fedor) [inline anmykh/checkio/checkio_2.py:10]: a * b

File: tests/anmykh/test_homework_1.py [hunk 12]
```diff
@@ -0,0 +1,14 @@
+import pytest
+
+
+@pytest.mark.parametrize(["item", "data_type"],
+                         [("sad", str)])
+def test_is_string(item, data_type):
+    assert isinstance(item, data_type)
```
Comment (kudinov-fedor) [inline tests/anmykh/test_homework_1.py:7]: more readable:

```
@pytest.mark.parametrize(["item", "data_type"], [
    ("sad", str)
])
def test_is_string(item, data_type):
    assert isinstance(item, data_type)
```

File: tests/anmykh/test_homework_3.py [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    assert range(10) == range(0, 10, 1)
+    assert set() != ()
+    assert dict() == {}
+
+
+@pytest.mark.parametrize(["item1", "item2"],
+                         [([1, 2, 3], [1, 2, 3])])
+def test_is(item1, item2):
+    assert item1 == item2
+    assert item1 is not item2
```
Comment (kudinov-fedor) [inline tests/anmykh/test_homework_3.py:26]: no need to use parametrized if there is only 1 test

**Review by kudinov-fedor — APPROVED** [review]

File: tests/anmykh/test_homework_1.py (hunk 12 above)
Comment (fkudinov-mn) [reply tests/anmykh/test_homework_1.py:8]: this comment relates to all parametrizes not only to this one

File: tests/anmykh/test_homework_3.py (hunk 13 above)
Comment (fkudinov-mn) [reply tests/anmykh/test_homework_3.py]: this comment ralates to all such cases

**Review by kudinov-fedor — APPROVED** [review]

**Pull request merged**